src/Checkbox.js

```diff
@@ -0,0 +1,111 @@
+import React from 'react'
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+import Icon from './Icon'
+import Flex from './Flex'
+import Box from './Box'
+import theme from './theme'
+
+class Checkbox extends React.Component {
+  constructor({ value, isChecked }) {
+    super()
+    this.state = {
+      value,
+      isChecked
+    }
+  }
+
+  handleClick() {
+    const { isDisabled, name, value, checkAction } = this.props
+    const { isChecked } = this.state
+    if (isDisabled) {
+      return
+    }
+
+    checkAction(name, value, !isChecked)
+    this.setState({ isChecked: !isChecked })
+  }
+
+  render() {
+    const { name, id, isDisabled, value, size, fontSize } = this.props
+
+    const currentTheme = this.props.theme || theme
+
+    const CurrentComponent = this.state.isChecked
+      ? CheckedCheckbox
+      : StyledCheckbox
+    return (
+      <CurrentComponent
+        isChecked={this.state.isChecked}
+        isDisabled={isDisabled}
+        size={size}
+        theme={currentTheme}
+        data-test="design-system-checkbox"
+        onClick={() => this.handleClick()}
+      >
+        {this.state.isChecked ? (
+          <Icon
+            name="check"
+            size={size - 4}
+            color={currentTheme.colors.white}
+          />
+        ) : null}
+
+        <input
+          type="checkbox"
+          name={name}
+          id={id}
+          value={value}
+          disabled={isDisabled}
+          defaultChecked={this.state.isChecked}
+          style={{ display: 'none' }}
```

Comment:
For accessibility reasons, it's better to *not* set `display: none` on the underlying form element, instead, using `position: absolute` and `opacity: 0` is one way to accessibly hide form elements. An example can be seen here: http://wtfforms.com

Reply:
true, shame on me for not having accessibility in mind